### PR TITLE
doc: developer-documentation: Add example how to add a new kselftest

### PR DESCRIPTION
**Pull request by musamaanjum**

### CODE REVIEW

File: doc/developer-documentation.md [hunk 1]
```diff
@@ -170,6 +170,35 @@ Here, `node` refers to the name of API PubSub channel where node events are publ
 
 After following these steps, run your pipeline instance to activate your newly added test configuration.
 
+### Example to enable a new kselftest
```
Comment (JenySadadia) [inline doc/developer-documentation.md:173]: I'd suggest to make it generic:
```suggestion
### A complete example of enabling a new job
```

File: doc/developer-documentation.md [hunk 2]
```diff
@@ -170,6 +170,35 @@ Here, `node` refers to the name of API PubSub channel where node events are publ
...
+
+The kselftests (found in `<linux_kernel>/tools/testing/selftests`) are enabled suite by suite. Most parameters are already set for other kselftests. Those must be reused while enabling new kselftests which make enabling new tests pretty straight forward. The `dt` kselftest suite can be enabled by:
```
Comment (JenySadadia) [inline doc/developer-documentation.md:175]:
```suggestion
Here is an example of enabling `kselftest` (found in `<linux_kernel>/tools/testing/selftests`) job. Most parameters are already set for other kselftests. Those must be reused while enabling new kselftests which makes enabling new tests pretty straight forward. The `dt` kselftest suite can be enabled by:
```

File: doc/developer-documentation.md [hunk 3]
```diff
@@ -170,6 +170,35 @@ Here, `node` refers to the name of API PubSub channel where node events are publ
...
+
+1. Create a job definition in `jobs` section.
+```
+  kselftest-dt:
+    template: kselftest.jinja2
+    kind: job
+
```
Comment (JenySadadia) [inline doc/developer-documentation.md:182]: Please drop this empty line.

File: doc/developer-documentation.md [hunk 4]
```diff
@@ -170,6 +170,35 @@ Here, `node` refers to the name of API PubSub channel where node events are publ
...
+      nfsroot: 'http://storage.kernelci.org/images/rootfs/debian/bookworm-kselftest/20240221.0/{debarch}'
+      collections: dt
+      job_timeout: 10
+    rules:
+      min_version:
+        version: 6
+        patchlevel: 7
+    kcidb_test_suite: kselftest.dt
+```
+2. Create a `scheduler` entry for `kselftest-dt` job. Lets schedule this suite in `lava-collabora` lab at `kbuild-gcc-12-arm64` build completion event. The entry by using already defined anchors would be:
```
Comment (JenySadadia) [inline doc/developer-documentation.md:193]:
```suggestion
2. Create a `scheduler` entry for `kselftest-dt` job. Let's schedule this suite in `lava-collabora` lab at `kbuild-gcc-12-arm64` build completion event. The entry by using already defined anchors would be:
```

**Review by JenySadadia — CHANGES_REQUESTED** [review]
Some suggestions:

File: doc/developer-documentation.md (hunk 1 above)
Comment (musamaanjum) [reply doc/developer-documentation.md:173]: Changing it to "A example of enabling a new job"

File: doc/developer-documentation.md (hunk 4 above)
Comment (JenySadadia) [reply doc/developer-documentation.md:192]: This isn't resolved yet.

Comment (musamaanjum) [reply doc/developer-documentation.md:192]: Seems like there was issue with force push. Updated now.

**Review by JenySadadia — APPROVED** [review]
Thanks. LGTM.

**Comment by musamaanjum** [comment]
@nuclearcat @a-wai let's merge this PR.

**Pull request merged**